src/cpp/core/include/core/collection/Stack.hpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Stack.hpp
+ *
+ * Copyright (C) 2009-2015 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#ifndef CORE_STACK_HPP
+#define CORE_STACK_HPP
+
+#include <vector>
+
+namespace rstudio {
+namespace core {
+namespace collection {
+
+template <typename T>
+class Stack
```

Comment:
I did not have a strong reason beyond 1) preferring `vector` over `deque` as the underlying storage type, and 2) being able to refine / tweak the interface if needed.